core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -502,6 +503,8 @@ class ProducerStateManager(
   private var lastMapOffset = 0L
   private var lastSnapOffset = 0L
 
+  newGauge("ProducerIdCount", () => producers.size)
```

Comment:
I think the other concern with this approach is that we have this metric per ProducerStateManager (per partition!) when we really want it per broker. https://github.com/apache/kafka/pull/13078 resolves this.